lib/puppetserver/ca/local_certificate_authority.rb

```diff
@@ -37,6 +37,38 @@ class LocalCertificateAuthority
         ["authorityKeyIdentifier", "keyid:always", false]
       ].freeze
 
+      def self.puppet_confdir
+        if Puppetserver::Ca::Utils::FileSystem.running_as_root?
+          '/etc/puppetlabs/puppet'
+        else
+          "#{ENV['HOME']}/.puppetlabs/etc/puppet"
+        end
+      end
+
+      def self.puppetserver_confdir
+        if Puppetserver::Ca::Utils::FileSystem.running_as_root?
+          '/etc/puppetlabs/puppetserver'
+        else
+          "#{ENV['HOME']}/.puppetlabs/etc/puppetserver"
+        end
+      end
```

Comment:
Can we utilize these where Justin had similar logic?

Reply:
Like somehow consolidate with https://github.com/puppetlabs/puppetserver-ca-cli/blob/main/lib/puppetserver/ca/config/puppet.rb#L53-L74

Reply:
yeah, I was thinking about that. Wasn't sure how much we wanted to keep the config stuff as its own thing? but I can update.

Reply:
Could possibly put these in with the filesystem stuff and call them both places? Not totally sure what the right separation of concerns is, but the logic is finicky and hardcoded and I'd rather that only be in one place.

Reply:
fwiw, I think these methods are available on the "puppet" config instance here: https://github.com/puppetlabs/puppetserver-ca-cli/pull/65/files#diff-8a63d3dddb93d86b02aba1619d4d8e877d75584395f677c0dbbce246e40d0b81R58

Passing either the whole config instance (since the settings are also a member of it) or gather these in the caller would probably work?

Reply:
ugh diff links on PRs like that never work for me

Reply:
but I think I know where you're talking about

Reply:
They don't work for me either.

Reply:
hmm those puppet config methods are _not_ available in the tests.. Does it make sense to create an instance just to access those methods?

Reply:
okay I just updated Justin's code to use these methods.

Reply:
How would you feel about defining these in https://github.com/puppetlabs/puppetserver-ca-cli/blob/main/lib/puppetserver/ca/utils/config.rb as static functions? That feels like a better home for them than in this class. I don't _really_ like requiring this file in the config class (a little bit worried about winding up in a circular dependency by accident).

Reply:
sure

Reply:
done

Reply:
Thanks, I think that feels more right.